standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -326,10 +356,26 @@ void checkTable(Table table, PartitionIterable parts, byte[] filterExp, CheckRes
       CheckResult.PartitionResult prFromMetastore = new CheckResult.PartitionResult();
       prFromMetastore.setPartitionName(getPartitionName(table, partition));
       prFromMetastore.setTableName(partition.getTableName());
-      if (!fs.exists(partPath)) {
-        result.getPartitionsNotOnFs().add(prFromMetastore);
-      } else {
+      if (allPartDirs.contains(partPath)) {
```

Comment:
allPartDirs.remove(partPath); will return true if the element exits and false otherwise. we can have one statement instead of contains and then remove